prov/efa/src/efa_rma.c

```diff
@@ -61,15 +65,30 @@ ssize_t efa_rma_post_read(struct efa_ep *ep, const struct fi_msg_rma *msg, uint6
 	}
 
 	/* caller must provide desc because EFA require FI_MR_LOCAL */
-	assert(msg->desc && msg->desc[0]);
+	assert(msg->desc);
 
 	qp = ep->qp;
 	ibv_wr_start(qp->ibv_qp_ex);
 	qp->ibv_qp_ex->wr_id = (uintptr_t)msg->context;
-	ibv_wr_rdma_read(qp->ibv_qp_ex, msg->rma_iov[0].key, msg->rma_iov[0].addr);
 
-	ibv_mr = (struct ibv_mr *)msg->desc[0];
-	ibv_wr_set_sge(qp->ibv_qp_ex, ibv_mr->lkey, (uint64_t)msg->msg_iov[0].iov_base, msg->msg_iov[0].iov_len);
+	iov_count = msg->iov_count;
+	memcpy(iov, msg->msg_iov, iov_count * sizeof(struct iovec));
```

Comment:
I'm not following this logic here. libibverbs can handle a list of buffers via `ibv_wr_set_sge_list`, why not use that?

Reply:
ibv_wr_set_sge_list was intended for send. For rdma read, one will need to pair rma_iov and iov precisely like the following. Noticing ```ibv_wr_rdma_read()``` does not allow you to set length, because it assume caller will later call ```ibv_wr_set_sge``` to set length, which means ```iov``` length and ```rma_iov``` length must match

Reply:
`efa_send_wr_set_sge_list` (the function this points to) supports rdma read and can do multiple sge's.

In regards to rma_iov, I'd prefer that we just advertise an rma length of one. The verbs and gni providers do this.

Reply:
Thank you! I used ibv_wr_set_sge_list in new revision, and set info->tx_attr->rma_iov_limit to 1 in efa_fabric.